py-polars/tests/unit/datatypes/test_temporal.py

```diff
@@ -576,38 +579,119 @@ def test_upsample(time_zone: str | None, tzinfo: ZoneInfo | timezone | None) ->
         }
     ).with_columns(pl.col("time").dt.replace_time_zone(time_zone).set_sorted())
 
-    up = df.upsample(
-        time_column="time", every="1mo", group_by="admin", maintain_order=True
-    ).select(pl.all().forward_fill())
+    if offset is not None:
+        with pytest.warns(
+            DeprecationWarning,
+            match="`offset` is deprecated and will be removed in the next breaking release. "
+            "Instead, chain `upsample` with `.dt.offset_by`.",
+        ):
+            up = df.upsample(
+                time_column="time",
+                every="1mo",
+                group_by="admin",
+                maintain_order=True,
+                offset=offset,
+            ).select(pl.all().forward_fill())
+    else:
+        up = df.upsample(
+            time_column="time",
+            every="1mo",
+            group_by="admin",
+            maintain_order=True,
+        ).select(pl.all().forward_fill())
```

Comment:
a pattern that's used elsewhere in tests is something like

```python
    context_manager: contextlib.AbstractContextManager[pytest.WarningsRecorder | None]
    if offset is not None:
        context_manager = pytest.deprecated_call(match=msg)
    else:
        context_manager = contextlib.nullcontext()
```
and then just put the `up = df.upsample` under the `context_manager` context